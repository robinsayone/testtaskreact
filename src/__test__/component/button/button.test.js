import React from 'react';
import ReactDom from 'react-dom';
import UiButton from './../../../component/button/button';
import { render,cleanup , screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

afterEach(cleanup); //its cleans test after every test

it("render without crashing",()=>{
    const div = document.createElement("div");
    ReactDom.render(<UiButton></UiButton>,div);
});

it("renders button correctly",()=>{
    render(<UiButton buttontext="testButton" className="testClass"></UiButton>);
    expect(screen.getByTestId('uiButton')).toHaveTextContent("testButton");

});

it("matches snapshot",()=>{
    const tree = renderer.create(<UiButton buttontext="save"></UiButton>).toJSON();
    expect(tree).toMatchSnapshot();
});