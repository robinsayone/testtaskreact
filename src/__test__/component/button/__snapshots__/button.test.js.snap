// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
  data-testid="uiButton"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  save
</button>
`;
